### PR TITLE
Feature: Initial screens

### PR DESCRIPTION
**Pull request by Charan017**
- Implemented Screens
  - PostsScreen
  - BookmarksScreen
  - UpdateProfileScreen
  - CreatePostScreen
  - SettingsScreen
 - Implemented SDK changes

### CODE REVIEW

File: android/gradle/wrapper/gradle-wrapper.properties [hunk 1]
```diff
@@ -1,5 +1,6 @@
...
-distributionPath=wrapper/dists
-distributionUrl=https\://services.gradle.org/distributions/gradle-7.4.2-all.zip
-zipStoreBase=GRADLE_USER_HOME
-zipStorePath=wrapper/dists
+#Wed Aug 03 15:32:22 IST 2022
+distributionBase=GRADLE_USER_HOME
+distributionUrl=https\://services.gradle.org/distributions/gradle-7.3.3-bin.zip
+distributionPath=wrapper/dists
+zipStorePath=wrapper/dists
+zipStoreBase=GRADLE_USER_HOME
```
Comment (ravitilicho) [inline android/gradle/wrapper/gradle-wrapper.properties:6]: @Charan017 please fix the ^M (Ctrl + M) as new lines as per GitHub recommendation using this solution: https://stackoverflow.com/a/1889699

File: package.json [hunk 2]
```diff
@@ -10,22 +10,25 @@
     "lint": "eslint ."
   },
   "dependencies": {
+    "@dashx/react-native": "file:../dashx-react-native",
```
Comment (ravitilicho) [inline package.json]: @Charan017 please remove this local repo reference. Please point to an appropriate version of `@dashx/react-native` repo

File: src/components/ActionSheet.js [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+        setDisplayActionSheet(false);
+        console.log('Upload error: ', error);
+        showToast(`Upload asset error: ${JSON.stringify(error)}`);
+      }
+    } else {
+      try {
+        if (await checkStorageReadPermission()) {
+          const {
+            assets: [imagePickerResponse],
+          } = await actions[index]?.({mediaType: 'video', includeBase64: true});
```
Comment (ravitilicho) [inline src/components/ActionSheet.js]: @Charan017 Why are we using `includeBase64: true` here? Are we using file's base64 string anywhere during our processing?

File: src/components/CreatePost.js [hunk 4]
```diff
@@ -0,0 +1,237 @@
...
+  const [storePostImage, setStorePostImage] = useState();
+  const [storePostVideo, setStorePostVideo] = useState();
+  const [mediaType, setMediaType] = useState('');
+  const [showActionSheet, setShowActionSheet] = useState(false);
+
+  const validation = () => {
+    let count = 0;
+
+    for (let key in errorMessage) {
+      let validationResponse = validate(key, eval(key));
```
Comment (ravitilicho) [inline src/components/CreatePost.js:39]: @Charan017 let's move away from using `eval` for validation across all the forms we have. It's not recommended pattern to follow. Please create a ticket for it to track.

File: src/components/CreatePost.js [hunk 5]
```diff
@@ -0,0 +1,237 @@
...
+  };
+
+  const onPickPostImage = async pickedMedia => {
+    setShowActionSheet(false);
+
+    try {
+      if (mediaType === 'image') {
+        const response = await DashX.uploadExternalAsset(
+          pickedMedia,
+          'e8b7b42f-1f23-431c-b739-9de0fba3dadf',
```
Comment (ravitilicho) [inline src/components/CreatePost.js]: @Charan017 Please store these in a constants file and reference it from there.

File: src/components/ImageAndVideoModal.js [hunk 6]
```diff
@@ -0,0 +1,82 @@
...
+  StatusBar,
+  StyleSheet,
+  TouchableOpacity,
+  View,
+} from 'react-native';
+import Video from 'react-native-video';
+
+export const ImageAndVideoModal = ({visible, setModal, imageOrVideoUri}) => {
+  return (
+    // <StatusBar backgroundColor={'black'}>
```
Comment (ravitilicho) [inline src/components/ImageAndVideoModal.js:7]: @Charan017 Please remove this comment if unnecessary.

File: src/components/createPosts.js [hunk 7]
```diff
@@ -0,0 +1,144 @@
...
+} from 'react-native';
+import InputText from './inputText';
+import Button from './button';
+import AppContext from '../useContext/AppContext';
+import ShowError from './showError';
+import validate from './validator';
+import {showToast} from '../utils/LocalStorage';
+import {postmethod} from '../utils/LocalStorage';
+
+const CreatePosts = ({visible, handleModal, getAllPosts}) => {
```
Comment (ravitilicho) [inline src/components/createPosts.js]: @Charan017 ~CreatePosts~ -> `CreatePostModal` for the purpose clarity.

File: src/components/showPosts.js [hunk 8]
```diff
@@ -0,0 +1,118 @@
+import React, {useContext} from 'react';
+import {View, Text, Image, StyleSheet, TouchableOpacity} from 'react-native';
+import moment from 'moment';
+import AppContext from '../useContext/AppContext';
+import {putMethod} from '../utils/LocalStorage';
+import {getAllPosts} from '../utils/LocalStorage';
+import {showToast} from '../utils/LocalStorage';
+
+const ShowPosts = ({item, setIsModalVisible, setGetPost, getUserbookmarks}) => {
```
Comment (ravitilicho) [inline src/components/showPosts.js]: @Charan017 ~ShowPosts~ -> `PostsList`

File: src/routes/tabs.js [hunk 9]
```diff
@@ -24,26 +24,79 @@ const Tabs = () => {
...
-        options={{headerShown: false}}
+        options={{
+          headerShown: false,
+          tabBarIcon: ({focused}) => {
+            return (
+              <View>
+                <Image
+                  source={require('../assets/store.png')}
+                  resizeMode="contain"
+                  style={{width: 25, tintColor: focused ? 'blue' : null}}
```
Comment (ravitilicho) [inline src/routes/tabs.js]: @Charan017 Please use `colors.js` or similar for all the color constants for reusability.

**Review by ravitilicho — CHANGES_REQUESTED** [review]

File: src/components/showPosts.js (hunk 8 above)
Comment (Charan017) [reply src/components/showPosts.js]: Done

File: src/components/createPosts.js (hunk 7 above)
Comment (Charan017) [reply src/components/createPosts.js]: Done

File: src/components/ImageAndVideoModal.js (hunk 6 above)
Comment (Charan017) [reply src/components/ImageAndVideoModal.js:7]: Done

File: src/components/CreatePost.js (hunk 5 above)
Comment (Charan017) [reply src/components/CreatePost.js]: Done

File: src/components/ActionSheet.js (hunk 3 above)
Comment (Charan017) [reply src/components/ActionSheet.js]: By removing `includeBase64` it's working fine. Actually it is not required.

File: src/components/CreatePost.js (hunk 4 above)
Comment (Charan017) [reply src/components/CreatePost.js:39]: Done

File: src/routes/tabs.js (hunk 9 above)
Comment (Charan017) [reply src/routes/tabs.js]: Done

File: package.json (hunk 2 above)
Comment (Charan017) [reply package.json]: Done

File: android/gradle/wrapper/gradle-wrapper.properties (hunk 1 above)
Comment (Charan017) [reply android/gradle/wrapper/gradle-wrapper.properties:6]: Done

**Pull request merged**